src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
     DefaultSceneHandlerImpl() = default;
     ~DefaultSceneHandlerImpl() override{};
 
+    /// @brief Encodes an attribute value list into a TLV structure and resizes the buffer to the size of the encoded data
+    /// @param aVlist[in] Attribute value list to encode
+    /// @param serializedBytes[out] Buffer to fill from the Attribute value list in a TLV format
+    /// @return CHIP_ERROR
+    virtual CHIP_ERROR
+    EncodeAttributeValueList(const app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> & aVlist,
+                             MutableByteSpan & serializedBytes)
+    {
+        TLV::TLVWriter writer;
+        TLV::TLVType outer;
+        writer.Init(serializedBytes);
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer));
```

Comment:
I have to ask: why is this outer structure envelope needed?  Is the idea to allow extensibility later?  It might be worth documenting this part.

Reply:
This (and the resulting failure when trying to encode without it) was my interpretation of Matter main spec:

> 5.1.5. TLV Content
> [...]All elements SHALL be housed within an anonymous top-level structure container.

Then in List.h, the encode function will encode a TLV::kTLVType_Array, which, when associated with an TLV::AnonymousTag(), caused WriteElementHead to fail.

So it lead me to believe than one cannot encode a List without an outer structure container.

Reply:
5.1.5 is about the TLV parts of a commissioning payload.  That's not a general TLV requirement.

> The encode function will encode a TLV::kTLVType_Array, which, when associated with an TLV::AnonymousTag(), caused WriteElementHead to fail

That's odd.  I would expect this to work fine....  If you look at `TLVPacketBufferBackingStoreTest::BasicEncodeDecode` is uses a toplevel anonymous-tagged TLV array for that test, and it works fine.

Reply:
Indeed, I think that I must have tried to keep the data model type `app::Clusters::Scenes::Structs::ExtensionFieldSet::Fields::kAttributeValueList` for some obscure reason.

Regardless, it works fine without it so I removed it.